procrustes/psdp.py

```diff
@@ -23,18 +23,249 @@
+
+    unpad_col : bool, optional
+        If True, zero columns (with values less than 1.0e-8) on the right-hand side of the intial
+        :math:`\mathbf{A}` and :math:`\mathbf{B}` matrices are removed.
+
+    unpad_row : bool, optional
+        If True, zero rows (with values less than 1.0e-8) at the bottom of the intial
+        :math:`\mathbf{A}` and :math:`\mathbf{B}` matrices are removed.
+
+    check_finite : bool, optional
+        If True, convert the input to an array, checking for NaNs or Infs.
+
+    weight : ndarray, optional
+        The 1D-array representing the weights of each row of :math:`\mathbf{A}`. This defines the
+        elements of the diagonal matrix :math:`\mathbf{W}` that is multiplied by :math:`\mathbf{A}`
+        matrix, i.e., :math:`\mathbf{A} \rightarrow \mathbf{WA}`.
+
+    Returns
+    -------
+    ProcrustesResult
+        The result of the Procrustes transformation.
+
+    Notes
+    -----
+    The OptPSDP algorithm (on which this implementation is based) is defined well in p. 114
+    of [1]_.
+
+    References
+    ----------
+    .. [1] Harry F. Oviedo (2019). "A Spectral Gradient Projection Method for the Positive
+        Semi-definite Procrustes Problem". Revista Colombiana de Matematicas, Volume 55(2021)1,
+        pages 109-123.
+    """
+    a, b = setup_input_arrays(
+        a,
+        b,
+        unpad_col,
+        unpad_row,
+        pad,
+        translate,
+        scale,
+        check_finite,
+        weight,
+    )
+    if a.shape != b.shape:
+        raise ValueError(
+            f"Shape of A and B does not match: {a.shape} != {b.shape} "
+            "Check pad, unpad_col, and unpad_row arguments."
+        )
+
+    # Initializing the required minimizer.
+    # Here, a denote the matrix to be transformed i.e. A, x is the transformer X
+    # and b is the target matrix B. Our goal is to minimize ||XA - B||_F
+    # (as mentioned in the function description).
+    n, m = a.shape
+    x = np.eye(n)
+
+    # Option structure with fields that serve as parameters for the algorithm.
+    if options is None:
+        options = {
+            # Maximum number of iterations.
+            "mxitr": 10000,
+            # Stop control for ||X_k - X_{k-1}||_F.
+            "xtol": 1e-5,
+            # Stop control for |F_k - F_{k-1}|/(1+|F_{k-1}|).
+            "ftol": 1e-12,
+            # If proj is True we perform Cholesky decomposition else we do spectral
+            # decomposition.
+            "proj": True,
+            # Parameter of the non-monotone technique proposed by Zhang-Hager.
+            "gamma": 0.85,
+            # Parameter for control the linear approximation in line search.
+            "rho": 1e-4,
+            # Factor for decreasing the step size in the backtracking line search.
+            "eta": 0.1,
+            # Initial step size.
+            "tau": 1e-3,
+            # Number of internal iterations.
+            "nls": 5,
+        }
```

Comment:
It would be problematic if a user only defines part of the dictionary. So, move this to the default argument. Thanks.

Reply:
Yep, I will update this right away.

Reply:
fixed in [118fa23](https://github.com/theochem/procrustes/pull/195/commits/118fa237e6143020fa876793cfc3b17c4f780a46)